pkg/bloomgateway/bloomgateway.go

```diff
@@ -315,21 +327,63 @@ func (g *Gateway) FilterChunkRefs(ctx context.Context, req *logproto.FilterChunk
 		return req.Refs[i].Fingerprint < req.Refs[j].Fingerprint
 	})
 
-	task, resCh, errCh, err := NewTask(tenantID, req)
+	var expectedResponses int
+	swbPerDay := partitionRequest(req)
 
-	if err != nil {
-		return nil, err
+	tasks := make([]Task, 0, len(swbPerDay))
+	for _, seriesWithBounds := range swbPerDay {
+		task, err := NewTask(tenantID, seriesWithBounds, req.Filters)
+		if err != nil {
+			return nil, err
+		}
+		tasks = append(tasks, task)
+		expectedResponses += len(seriesWithBounds.series)
+	}
+
+	if len(tasks) == 0 {
```

Comment:
I think we can have this check with `swbPerDay` right after `partitionRequest` (not a big deal tho)